CD 804xx relnotes CDS-7758 (#3036)
harness/developer-hub: release-notes/continuous-delivery.md

<Kustomizedep />

## Latest: Version 804xx

### New features and enhancements

This release does not have new features.

### Early access features

This release does not include early access features.
  
### Fixed issues

- Fixed an issue that caused the **Pipeline Executions** page to make unnecessary API calls to a back-end service. (CDS-77401)

- Fixed an issue where users could not specify runtime inputs for some advanced settings (such as Looping Strategy) in step groups and step-group templates. With this fix, runtime inputs are supported for all advanced step-group settings. (CDS-77246, ZD-49339, ZD-49427)

- Fixed a UI issue that caused some edit buttons to not appear. (CDS-76977)

- Fixed an issue where the K8s Apply step did not correctly evaluate expressions for the **Command Type** field. (CDS-76632)

- Fixed an issue where the reconciliation for input set templates did not check for differences in old vs. new YAML for a given input set. This was happening because there was no flag to monitor for changes in YAML strings. (CDS-76533).

>   This fix introduces a new `yamlDiffPresent` boolean in `InputSetYamlDiffDTO` and its relevant unit tests. The purpose of this flag is similar to validYAML, which determines whether the current YAML is valid or not.

Reviewer: This description does not have a Jira ID. Also, it mentions unit tests. I suspect it is exposed to users, but the mention of unit tests and the absence of a Jira ticket aroused my curiosity. Should we check? If it impacts users, like I think it does, we might have to rephrase it a bit.

Author: This is one paragraph in a two-paragraph summary. Sandesh has approved my [summary draft](https://harness.atlassian.net/browse/CDS-76533?focusedCommentId=560120), I'm going to leave this as-is.

- Fixed a UI issue where fields in Add Stage did not reposition correctly when the window size was decreased (CDS-76431).

- Fixed an issue where characters could get arranged incorrectly when updating an expression within a string value. For example, updating the expression in `test/var/<+expression.name>/login` might result in `test/var/<+expression.name`**/>**`login`. This fix ensures that expressions within strings are updated correctly.  (CDS-76354, ZD-48515)

- Fixed an issue where users could not save a Manual Approval step as a template when `autoApprove : action` is set to `APPROVE` or `REJECT`.  (CDS-76350)

- Added an information banner and doc link to the **Synchronization Barrier** step that highlights best practices when synchronizing deployments using barriers. (CDS-76291, ZD-48636)

- Fixed a UI issue where the **Environment** tab in a Deploy stage did not display new custom variables after they were created.  (CDS-76068)

- Removed the **Reconcile** option for individual input sets in the **Input Sets** page. (CDS-75845)

> - Implemented a fix to ensure that user-entered values are always populated in the **Inputs** tab on running / re-running the pipeline. (CDS-75593, ZD-48181)

Reviewer: ```suggestion
- Implemented a fix to ensure that user-entered values are always populated on the **Inputs** tab when running or rerunning a pipeline. (CDS-75593, ZD-48181)
```

> - Fixed a filtering issue in the **Pipeline Executions** page. (CDS-75224)

Reviewer: ```suggestion
- Fixed a filtering issue on the **Pipeline Executions** page. (CDS-75224)
```

>   When the service received empty arrays for fields in the filter, the service would add this fields in the filtering criteria with empty values thereby not returning accurate results.

Reviewer: ```suggestion
  If the service received empty arrays for fields in a filter, it added the fields to the filtering criteria with empty values, thereby not returning accurate results.
```
Better still (I'll begin my desc with "Services". Rewrite to "The service: if necessary, but we might have to clarify what service we are referring to):
> Services added fields to the filtering criteria even if they received those fields with empty arrays. Consequently, accurate results were not returned.

>   The fix will handle the empty arrays being sent and saved filters will return only fields that have values in them.

Reviewer: ```suggestion
  The fix handles empty arrays, and saved filters return only fields that have values.
```

> - Fixed an issue where ECR image based triggers were firing off with null values for artifact image tag and repo name. (CDS-75173)

Reviewer: ```suggestion
- Fixed an issue where triggers based on ECR images were fired with null values for artifact image tag and repo name. (CDS-75173)
```
Should it be "even when the artifact image tag and repository name had null values"?

> - Added a tooltip for Build stages to let users know `CI Stage can be skipped with New Artifact/Manifest Trigger using selective stage configuration`. (CDS-75080, ZD-47902) 

Reviewer: ```suggestion
- Added the following tooltip for the Build stage: `CI Stage can be skipped with New Artifact/Manifest Trigger using selective stage configuration`. (CDS-75080, ZD-47902) 
```
